torch/csrc/jit/python/pybind_utils.h

```diff
@@ -144,6 +144,16 @@ struct VISIBILITY_HIDDEN PythonFutureWrapper
         PyObjectType::get()));
   }
 
+  void add_done_callback(py::function cb) {
+    auto pf = std::make_shared<PythonFunctionGuard>(std::move(cb));
+    fut->addCallback(std::bind(
+        [pyFut(this->getPtr())](std::shared_ptr<PythonFunctionGuard> pf) {
+          pybind11::gil_scoped_acquire ag;
+          pf->func_(pyFut);
```

Comment:
what if this throws error? can we add a test to verify the behavior?

Reply:
Done, thanks for the reminder (had to double check the API). Added logic to swallow and print any exceptions thrown in the callback + a test.